### PR TITLE
[Synthetics] Add support for global variables

### PR DESCRIPTION
**Pull request by romainberger**
Add support for synthetics' global variables.

It also updates the go client since it needed a new endpoint.

### CODE REVIEW

File: datadog/resource_datadog_synthetics_global_variable.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+func resourceDatadogSyntheticsGlobalVariableRead(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV1 := providerConf.DatadogClientV1
+	authV1 := providerConf.AuthV1
+
+	syntheticsGlobalVariable, _, err := datadogClientV1.SyntheticsApi.GetGlobalVariable(authV1, d.Id()).Execute()
+
+	if err != nil {
+		if strings.Contains(err.Error(), "404 Not Found") {
```
Comment (therve) [inline datadog/resource_datadog_synthetics_global_variable.go:69]: You should be able to use the HTTP response instead and check the status code. I know we used that pattern elsewhere, but it's super awkward.

File: datadog/resource_datadog_synthetics_global_variable.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (therve) [inline datadog/resource_datadog_synthetics_global_variable.go:26]: This is marked as optional in the doc, which seems to make sense to me.

File: datadog/resource_datadog_synthetics_global_variable.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		Read:   resourceDatadogSyntheticsGlobalVariableRead,
+		Update: resourceDatadogSyntheticsGlobalVariableUpdate,
+		Delete: resourceDatadogSyntheticsGlobalVariableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (therve) [inline datadog/resource_datadog_synthetics_global_variable.go:22]: Maybe we want to check it's not an empty string?

File: datadog/resource_datadog_synthetics_global_variable.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+			},
+			"tags": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (therve) [inline datadog/resource_datadog_synthetics_global_variable.go:35]: Do you support empty strings here as well?

File: datadog/resource_datadog_synthetics_global_variable.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tags": {
+				Type:     schema.TypeList,
+				Required: true,
```
Comment (therve) [inline datadog/resource_datadog_synthetics_global_variable.go:30]: Is it really required? I would assume an empty list works.

**Review by therve — CHANGES_REQUESTED** [review]
Looking good, some minor comments about the schema.

File: datadog/resource_datadog_synthetics_global_variable.go (hunk 5 above)
Comment (romainberger) [reply datadog/resource_datadog_synthetics_global_variable.go:30]: Yes it works with an empty list, but it needs to be set (like for synthetics tests). I suppose we could default to an empty list if nothing is specified in the config.

File: datadog/resource_datadog_synthetics_global_variable.go (hunk 4 above)
Comment (romainberger) [reply datadog/resource_datadog_synthetics_global_variable.go:35]: For the value yes, it is supported.

**Review by therve — APPROVED** [review]

**Pull request merged**